Argo data submission wf@0.2.0

edsu7 opened the pull request:
Testing scenario: `local-test-job-InputCram.json`
- Has a mixture of `FASTQ` and `CRAM` files for which `CRAM` should trigger a separate workflow
- to run:
```
curl ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa.gz -o hs37d5.fa.gz
samtools faidx hs37d5.fa.gz
nextflow run $(pwd)/../main.nf -params-file  $(pwd)/input/local-test-job-InputCram.json --ref_genome_fa $(pwd)/hs37d5.fa
```

Test scenario : `local-test-job-InputBam.json` and `local-test-job-InputFastq.json`  should also be run to observe that `cleanup` will not be triggered (b/c everything is local)

edsu7 committed:
[wfpm v0.8.0] started a new version argo-data-submission-wf@0.2.0 from argo-data-submission-wf@0.1.0 which was released

edsu7 committed:
initial commit of cram2bam(missing latest seq-tools pkg)

- altered `main.nf` to handle cram2bam
- test cases `local-test-job-InputCram.json` and supporting files
- fixed input fastqs
- added `cram2bam@0.1.0`,`payload-gen-seq-experiment@0.8.0`
- removal of older dependencies

edsu7 committed:
Update to seq-tools

lindaxiang commented on argo-data-submission-wf/main.nf:
```diff
@@ -114,11 +121,36 @@ upload_params = [
   *:(params.upload ?: [:])
 ]
 
+process hello_cram {
```
what's the usage of this process?

edsu7 replied:
Oops. Placeholder functions for testing. I'll remove

lindaxiang commented on argo-data-submission-wf/main.nf:
```diff
@@ -114,11 +121,36 @@ upload_params = [
+}
+
+process hello_bam {
+  input:
```
same question as above.

edsu7 replied:
https://github.com/icgc-argo/argo-data-submission/pull/42#discussion_r1004923216

lindaxiang commented on argo-data-submission-wf/main.nf:
```diff
@@ -158,36 +181,91 @@ workflow ArgoDataSubmissionWf {
+
+    // If reference genome is not provided...
+    if (ref_genome_fa.startsWith("NO_FILE")){
+      // Generate metadata payload per normal
```
The workflow depends on the value of `ref_genome_fa` to decide whether do cram2bam or not.
Do you think it is better to have a dedicated param flag like `cram_support` to indicate that the input files contain CRAMs. So that we can check at the beginning:
- if `cram_support=true`, `cram_sequence_files` is not empty && `ref_genome_fa` is not `NO_FILE`
- if `cram_support=false`, `cram_sequence_files` is empty

edsu7 replied:
Might make the command unnecessarily bulky with the addition of both `ref_genome_fa` and a `cram_support` flag. 

Instead I can add the `ref_genome_fa` check earlier so that the pipeline will exit if `ref_genome_fa` does not exist and `CRAM` files are detected?

lindaxiang replied:
Yeah, that's another solution. It's better to add some checks to avoid the scenario that the user forget to add `ref_genome_fa` or provide it without purpose, however this will mislead the running of the workflow, and failed at later steps.

edsu7 committed:
Update main.nf

- added flag to identify cram2bam conversion
- remove place holder functions
- added checks that `convert_cram` and `ref_genome_fa` must be submitted together; otherwise exit

edsu7 committed:
update main.nf

- added checkReference process to handle cram detection + `ref_genome_fa` flag handling

edsu7 committed:
Update main.nf

- Fixed file brace for `ref_genome_fa`

edsu7 committed:
Update main.nf

- correction to code regarding secondary reference files retrieval

edsu7 committed:
update `tsv`s and `json`s

- updated various examples to point to `submission-qa`

lindaxiang reviewed (approved):
Test and lgtm.
